gno.land/pkg/gnoclient/util.go

```diff
@@ -1,10 +1,10 @@
 package gnoclient
 
 func (cfg BaseTxCfg) validateBaseTxConfig() error {
-	if cfg.GasWanted < 0 {
+	if cfg.GasWanted <= 0 {
 		return ErrInvalidGasWanted
 	}
-	if cfg.GasFee < "" {
```

Comment:
I have no clue. It seems that the tests work when running them in parallel, but not when they are run separately. Super weird, as everything is declared in its own scope and there should be no shared memory between the error test cases. I have to look into it, it seems that the mock implementations might have an issue. Any clues?

Reply:
We figured it out - the tests that should've caught this were run in Parallel and didn't work properly because I didn't add a redeclaration of the testcase within the running loop. [Here's](https://github.com/leohhhn/gno/blob/1ea0510da392b05ca9d01d9911e3646c2a1fe3fe/gno.land/pkg/gnoclient/client_test.go#L691) what I'm talking about. [More info](https://go.dev/doc/faq#closures_and_goroutines).